src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -121,8 +121,16 @@ public RMProfileClient(AzureRMProfile profile)
                         newTenant == null &&                         
                         TryGetTenantSubscription(token, account, environment, tenant, subscriptionId, subscriptionName, out tempSubscription, out tempTenant))
                     {
-                        newTenant = tempTenant;
-                        newSubscription = tempSubscription;
+                        if (tempSubscription == null && i + 1 < tenants.Count())
```

Comment:
No need for the empty bock. You can invert the if to work for one case. Doesn't this work?
```
If( tempSubscription != null || i == (tenants.Count() -1) ) 
```